### PR TITLE
Closer integrate scope graphs, NaBL2 analysis, and Spoofax Core.

### PR DESCRIPTION
**Pull request by hendrikvanantwerpen**
Add context for scope graphs, and an analyzer to be used for NaBL2.

### CODE REVIEW

File: org.metaborg.spoofax.meta.core/src/main/resources/org/metaborg/spoofax/meta/core/generator/general/langspec/metaborg.yaml [hunk 1]
```diff
@@ -14,10 +14,6 @@ dependencies:
 {{#analysisNabl2}}  - org.metaborg:org.metaborg.meta.nabl2.shared:${metaborgVersion}
   - org.metaborg:org.metaborg.meta.nabl2.runtime:${metaborgVersion}
 {{/analysisNabl2}}
-pardonedLanguages:
```
Comment (Gohla) [inline org.metaborg.spoofax.meta.core/src/main/resources/org/metaborg/spoofax/meta/core/generator/general/langspec/metaborg.yaml]: Why was this removed? These languages give false positives that should be pardoned.

Comment (hendrikvanantwerpen) [inline org.metaborg.spoofax.meta.core/src/main/resources/org/metaborg/spoofax/meta/core/generator/general/langspec/metaborg.yaml]: At least for ESV and SDF3, I never pardon them, and do not have problems with that. I have the impression that the false positives happen in few projects only, and I miss actual errors when I don't realize they are pardoned. I'm not sure about Stratego actually, because I usually don't have it as an explicit dependency. Maybe these false positives are more common than I thought, then I'll put it back.
